test/marathon_client_test.exs

```diff
@@ -0,0 +1,33 @@
+defmodule MarathonEventExporter.MarathonClientTest do
+  use ExUnit.Case, async: true
+
+  alias MarathonEventExporter.{MarathonClient, SSEParser}
+
+  @doc """
+  Build a Marathon event from the given type and fields.
+
+  The :eventType field is always added/overridden, the :timestamp field is
+  added if one is not provided.
+  """
+  def marathon_event(event_type, fields) do
+    {:ok, data} = Map.new(fields)
+    |> Map.put_new(:timestamp, DateTime.utc_now |> DateTime.to_iso8601)
+    |> Map.put(:eventType, event_type)
+    |> JSX.encode
+    %SSEParser.Event{event: event_type, data: data}
+  end
+
+  test "stream_events streams events to a listener process" do
+    {:ok, fm} = start_supervised(FakeMarathon)
+    base_url = FakeMarathon.base_url(fm)
+    {:ok, _} = MarathonClient.stream_events(base_url, [self()])
+    # Stream an event, assert that we receive it within a second.
+    event = marathon_event("event_stream_attached", remoteAddress: "127.0.0.1")
+    FakeMarathon.event(fm, event.event, event.data)
+    assert_receive {:sse, ^event}, 1_000
```

Comment:
A second seems a bit long, but not sure what a good value would be.

Reply:
This is a test timeout, so it must be long enough to be robust and short enough to not make us wait too long for the results.

The common case is that the event is either already waiting or will be delivered almost immediately, so we're unlikely to wait very long unless we fail. On the other hand, if we're running tests on a slow system we don't want to give up on the event delivery too soon and have to rerun the tests because of spurious failures.

That said, one second it a number I made up because it "seemed reasonable". There wasn't any science or measurement behind it and I don't think it's really worth the effort to do any of that. I'm happy to change to a different number if you'd like to suggest one, though. :-)